app/controllers/upvotes_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UpvotesController < ApplicationController
+  def create
+    if user_signed_in?
+      @review = Review.find(params[:review_id])
+      @beach = @review.beach
+      @upvote = Upvote.new(
+        review_id: @review.id,
+        user_id: current_user.id
+      )
+      @review.score += 1
```

Comment:
This works, but ideally this logic should be refactored away out of the controller. Like refactor a method onto your `Review` class. `@review.upvote`

Does your view require that all these variables (review, beach, upvote, etc) be defined as instance variables? You can just define them as locals if only your controller needs them.